plugins/UsersManager/Controller.php

```diff
@@ -183,6 +185,9 @@ public function userSettings()
         $view->userTokenAuth = Piwik::getCurrentUserTokenAuth();
         $view->ignoreSalt = $this->getIgnoreCookieSalt();
 
+        $newsletterSignupOptionKey = 'UsersManager.newsletterSignup.' . $userLogin;
+        $view->showNewsletterSignup = Option::get($newsletterSignupOptionKey) === false;
```

Comment:
We could also not show the box if `Piwik\SettingsPiwik::isInternetEnabled()` is disabled